sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/models/AmqpDataBodyTest.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test for {@link AmqpDataBody}.
+ */
+public class AmqpDataBodyTest {
+    private static final byte[] CONTENTS_BYTES = "Some-contents".getBytes(StandardCharsets.UTF_8);
+    private static final BinaryData DATA_BYTES = new BinaryData(CONTENTS_BYTES);
+
+    /**
+     * Verifies we correctly set values via constructor for {@link AmqpAnnotatedMessage}.
+     */
+    @Test
+    public void constructorValidValues() {
+        // Arrange
+        final List<BinaryData> binaryDataList = Collections.singletonList(DATA_BYTES);
```

Comment:
It's always a single item collection. Why not add another one even though our current implementation layer does not support it.